### ISSUE TITLE
OSM modifications GPX export: exchange name and cmt

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#5528]
Currently exporting OSM modifications as GPX file results in a XML structure similar to this one:

```
  <wpt lat="47.2905881" lon="9.8544402">
    <name>note create</name>
    <desc>id: -2</desc>
    <cmt>restaurant</cmt>
  </wpt>
  <wpt lat="47.2922801" lon="9.8541763">
    <name>note create</name>
    <desc>id: -3</desc>
    <cmt>bench</cmt>
  </wpt>
  <wpt lat="47.2913831" lon="9.8492646">
    <name>note create</name>
    <desc>id: -4</desc>
    <cmt>fuel station</cmt>
  </wpt>
  <wpt lat="47.2878724" lon="9.8488457">
    <name>note create</name>
    <desc>id: -5</desc>
    <cmt>traffic lights</cmt>
  </wpt>
```

The `name` isn't very useful. The important information is stored in the `cmt` tag. I suggest to exchange this two tags to store the important information in the `name` tag in order to make the exported GPX files more useful for JOSM.

Here is why. Opening a GPX file with the current format in JOSM looks like this:
![josm_01_gpx](https://user-images.githubusercontent.com/1040784/40805701-63c80998-651f-11e8-81a0-0f7856d7d059.png)
With the new format it would look like this:
![josm_02_gpx](https://user-images.githubusercontent.com/1040784/40805707-68d7763a-651f-11e8-881e-f09c3f131dcf.png)

This is a minor improvement. The important information comes at the beginning rather at the end.

However sometimes it make sense to convert the GPX layer into a data layer. This way you can copy the contents of tags in JOSM, e.g. a restaurant name. Converting a GPX file with the current format in JOSM looks like this:
![josm_01_data](https://user-images.githubusercontent.com/1040784/40805803-bd322eb4-651f-11e8-81c7-7b210a938134.png)
With the new format it would look like this:
![josm_02_data](https://user-images.githubusercontent.com/1040784/40805808-c3dd8402-651f-11e8-8ab2-7be289241138.png)

This is much more useful than just seeing generic "note create" objects.

**Comment by vshcherb** [comment]
You definitely better use OSC format

**Comment by sonora** [comment]
Yes, that's how it should be, I think. Only thing needs testing is can wpt.name be null or empty under some circumstances, because it is a needed parameter for us, I think.

**Comment by scaidermern** [comment]
> You definitely better use OSC format

When trying to open this file, JOSM complains "No data found in file [...]". GPX is the preferred format for JOSM and other OSM editors.

**Comment by vshcherb** [comment]
That's a strange issue with OSC...

**Comment by scaidermern** [comment]
I don't know why opening of OsmAnd's OSC files in JOSM fails. OSC isn't supported by most OSM editors anyway whereas GPX can be opened in most if not all OSM editors (JOSM, iD, Vespucci, Potlatch, ...).

**Comment by scaidermern** [comment]
Looks like JOSM now supports OsmAnd's OSC files, see https://josm.openstreetmap.de/ticket/11086.
